From 3452ff6b53eb3638fb49dd2f1f75e64f9892cf51 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Wed, 24 Aug 2022 14:15:33 +0300
Subject: [PATCH] [SW-0] gaudi2: sram bist: add more error prints

This patch adds print in case ones and zeros check failures
with more information like read value, zeros/ones mask.

Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
Change-Id: I666b33edf7709e9d1a253cc122629c254738b0b1
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index 0700b1e..6d0ee4b 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -177,11 +177,15 @@
 static int gaudi2_sms_check_pattern(uint32_t value, uint32_t ones_mask, uint32_t zeros_mask)
 {
 	if ((value & ones_mask) != ones_mask) {
+		preboot_puts_hex("val: ", value, 0);
+		preboot_puts_hex("ones mask: ", ones_mask, 0);
 		preboot_puts("ones check failed\n");
 		return 1;
 	}
 
 	if ((~value & zeros_mask) != zeros_mask) {
+		preboot_puts_hex("val: ", value, 0);
+		preboot_puts_hex("zeros mask: ", zeros_mask, 0);
 		preboot_puts("zeros check failed\n");
 		return 1;
 	}
